indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -109,34 +124,151 @@ public static void setupClasspath(
     if (fs instanceof LocalFileSystem) {
       return;
     }
-
+    // Create distributed directory if it does not exist.
+    fs.mkdirs(distributedClassPath);
```

Comment:
this did not exist before, why is this needed now? and also why not fs.mkdirs(intermediateClassPath) too?